File: app/src/main/java/com/starsep/sokoban/mvc/GameController.java
package com.starsep.sokoban.mvc;

import android.content.Context;

import com.starsep.sokoban.gamelogic.Gameplay;

public interface GameController {
	void onStatsChanged();

	void onGamePause();

	void onSaveGame(Gameplay game);

	void onNewGame();

	boolean editMode();

	Context getContext();
}

File: app/src/main/java/com/starsep/sokoban/mvc/GameController.kt
package com.starsep.sokoban.mvc

import android.content.Context

import com.starsep.sokoban.gamelogic.Gameplay

interface GameController {

    val context: Context
    fun onStatsChanged()

    fun onGamePause()

    fun onSaveGame(game: Gameplay)

    fun onNewGame()

    fun editMode(): Boolean
}
